### PR TITLE
Removed stripNeverLearned from SP.compute and updated nupic.core

### LINKED ISSUE #1262: Remove stripNeverLearned from SP.compute
See also numenta/nupic.core#174

The `stripNeverLearned` parameter was added as a temporary measure to allow enabling a random SP. Remaining work:

- [ ] Update SP region to take `stripNeverLearned` parameter that defaults to `True`.
- [ ] Update examples and other clients to pass explicit values for `stripNeverLearned` and/or call `stripUnlearnedColumns` on results from `compute` method.
- [ ] Change `stripNeverLearned` defaults to `False`
- [ ] Remove `stripNeverLearned` parameter from SP.compute.

### LINKED ISSUE #2278: Remove stripNeverLearned parameter from spatial_pooler.compute
Partially replaces outdated issue #1262 

Separate ```stripUnlearnedColumns``` from ```compute``` in the spatial pooler and update  ```stripUnlearnedColumns``` to have same functionality as the same method in nupic.core.

### PR DESCRIPTION
**Pull request by jaredweiss**
Fixes #2278

Removed stripNeverLearned from ```SP.compute``` and updated ```SP.stripUnlearnedColumns``` to be consistent with nupic.core implementation.

@scottpurdy Please review

### CODE REVIEW

File: nupic/regions/SPRegion.py [hunk 1]
```diff
@@ -590,12 +590,13 @@ def _doBottomUpCompute(self, rfInput, resetSignal):
     inputVector = numpy.array(rfInput[0]).astype('uint32')
     outputVector = numpy.zeros(self._sfdr.getNumColumns()).astype('uint32')
 
-    # Switch to using a random SP if learning mode is off and the SP hasn't
-    # learned anything yet.
+    # Don't strip unlearned columns if learning is off and the SP hasn't
+    # learned anything yet. This acts as a random SP.
```
Comment (breznak) [inline nupic/regions/SPRegion.py]: haven't we got rid off `randomSP` recently? (I think `not learn` implied that)

File: nupic/research/spatial_pooler.py [hunk 2]
```diff
@@ -717,25 +710,26 @@ def compute(self, inputVector, learn, activeArray, stripNeverLearned=True):
...
     """Removes the set of columns who have never been active from the set of
     active columns selected in the inhibition round. Such columns cannot
     represent learned pattern and are therefore meaningless if only inference
     is required. This should not be done when using a random, unlearned SP
     since you would end up with no active columns.
 
-    @param activeColumns: An array containing the indices of the active columns
+    @param activeArray: An array whose size is equal to the number of columns.
+        Each column in this array will be checked to see if it has been active
+        before. If a column is active but it hasn't been learned, it will be
```
Comment (breznak) [inline nupic/research/spatial_pooler.py]: what does "column has been learned" mean? In relation to activeArray at times T, T-1

File: nupic/research/spatial_pooler.py [hunk 3]
```diff
@@ -717,25 +710,26 @@ def compute(self, inputVector, learn, activeArray, stripNeverLearned=True):
...
+        disabled.
     """
     neverLearned = numpy.where(self._activeDutyCycles == 0)[0]
```
Comment (breznak) [inline nupic/research/spatial_pooler.py]: I see answer to my Q above. Maybe better mention this "learned <=> activeDutyCycles > 0"

Comment (jaredweiss) [inline nupic/research/spatial_pooler.py]: Is this a better explanation? 
```An array whose size is equal to the number of columns. Any columns marked as active that have an activeDutyCycle of 0 have never been activated before and therefore are not active due to learning. Any unlearned columns will be disabled.```

Comment (breznak) [inline nupic/research/spatial_pooler.py]: do I understand correctly when I say this? 

```
An array whose size is equal to the number of columns. Any columns marked as active that have never been activated before (have 'activeDutyCycle' of 0) and therefore are have not been activated due to learning. Those (unlearned) columns will be disabled (set to 0). Basically that means ignoring columns that are activated for the first time. 
```

Comment (jaredweiss) [inline nupic/research/spatial_pooler.py]: Yes. That's the behavior of this method.

**Comment by scottpurdy** [comment]
:+1:

**Pull request merged**